packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -428,6 +428,8 @@ class _SnackBarState extends State<SnackBar> {
     final EdgeInsetsGeometry padding = widget.padding
       ?? EdgeInsetsDirectional.only(start: horizontalPadding, end: widget.action != null ? 0 : horizontalPadding);
 
+    final double actionHorizontalMargin = (widget.padding?.resolve(TextDirection.ltr).right ?? horizontalPadding) / 2;
```

Comment:
How did you calculate this? It looks like the spec shows a standard 8 pixels as the margin: https://material.io/components/snackbars#specs

Reply:
This was a little bit tricky. Spec indeed shows 8 pixels of margin here, but spec is only for a floating SnackBar. In the `SnackBar` we have a `padding` parameter which allows the developer to customize it. I figured out that in the spec, left padding of the whole snackbar is 16 px and right padding is 8px. But there is also a left padding for the button itself that's the same as this right padding - 8px. 16px on the left equals to 8px + 8px on the right. The screenshot below describes my idea behind that.

![image](https://user-images.githubusercontent.com/4085280/96037733-8114a900-0e66-11eb-90c6-264956e86da7.png)

Thanks to that even after providing a custom `padding` by the developer, our spacing will still be proportional to what the spec shows! :)

Reply:
Awesome! Thank you for the explanation. :)